modules/imap/functions.php

```diff
@@ -733,62 +733,66 @@ function sort_by_internal_date($a, $b) {
  * @return array
  */
 if (!hm_exists('merge_imap_search_results')) {
-function merge_imap_search_results($ids, $search_type, $session, $hm_cache, $folders = array('INBOX'), $limit=0, $terms=array(), $sent=false) {
+function merge_imap_search_results($ids, $search_type, $session, $hm_cache, $folders = array('INBOX'), $limit = 0, $terms = array(), $sent = false, $combined = false) {
```

Comment:
This function is already too complex, with eight parameters making its purpose unclear. This complexity will likely cause issues in the future, if not already, as larger functions are more prone to breaking in multiple parts of the application. It would be better to refactor it to ensure it serves a clear and focused purpose.

Reply:
@jacob-js This PR doesn't make the tag perfect, please feel free to change whatever you think should be changed for the greater good, including the `merge_imap_search_results`!